File: zendrive-api/src/main/java/com/zendrive/api/rest/model/dto/scan/ScanCheckResponse.java
package com.zendrive.api.rest.model.dto.scan;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder(builderMethodName = "Builder", builderClassName = "Builder", setterPrefix = "with")
public class ScanCheckResponse {
    private String path;
    private boolean exists;
    private long fileCount = 0;
    private long dirCount = 0;
    private long totalSize = 0;
    private String errorMessage = "";
}

File: zendrive-ui/src/api/models/scan/ScanCheckResponse.ts
class ScanCheckResponse {
	path: string;
	exists: boolean;
	fileCount: number;
	dirCount: number;
	totalSize: number;
	errorMessage: string;

	constructor(
		path: string,
		exists: boolean,
		fileCount: number,
		dirCount: number,
		totalSize: number,
		errorMessage: string
	) {
		this.path = path;
		this.exists = exists;
		this.fileCount = fileCount;
		this.dirCount = dirCount;
		this.totalSize = totalSize;
		this.errorMessage = errorMessage;
	}
}
